### ISSUE TITLE
Question: Reproduce data figure from paper

### ISSUE THREAD

**Issue by yarinbar** [soda-inria/caussim#1]
Hi Mat,
I am trying to reproduce the data generation process in order to achieve the same data distribution shown here:
![image](https://user-images.githubusercontent.com/37772157/218678463-519aa204-b55a-4835-879d-ff5a3439501b.png)

However when I wrote this code it doesnt change the between different seeds:
```python
from caussim.data.simulations import CausalSimulator
from caussim.experiences.base_config import DEFAULT_SIMU_CONFIG

from copy import deepcopy
import matplotlib.pyplot as plt
from matplotlib.pyplot import figure
import seaborn as sns
from sklearn import clone



if __name__ == '__main__':

    dataset_config = deepcopy(DEFAULT_SIMU_CONFIG)

    dataset_config["simu_random_seed"] = 0
    base_seed = 8
    overlap = 0.7
    n_components = 2
    dataset_config["treatment_ratio"] = 0.2
    dataset_config["treatment_link"]["params"]["random_state_feat"] = base_seed
    dataset_config["baseline_link"]["params"]["random_state_feat"] = base_seed
    dataset_config["outcome_noise"] = 0
    dataset_config["treatment_link"]["params"]["n_components"] = n_components
    dataset_config["baseline_link"]["params"]["n_components"] = n_components
    dataset_config["treatment_assignment"]["params"]["overlap"] = overlap
    n_samples = 1000

    sim = CausalSimulator(
        n_samples_init=dataset_config["n_samples_init"],
        dim=dataset_config["dim"],
        baseline_link=dataset_config["baseline_link"],
        effect_link=dataset_config["effect_link"],
        treatment_link=dataset_config["treatment_link"],
        treatment_assignment=dataset_config["treatment_assignment"],
        effect_size=dataset_config["effect_size"],
        outcome_noise=dataset_config["outcome_noise"],
        treatment_ratio=dataset_config["treatment_ratio"],
        random_seed=dataset_config["random_seed"],
        clip=dataset_config["clip"],
    )

    df = sim.sample().df

    graph = sns.jointplot(data=df, x='x_0', y='x_1', hue='a')
    plt.tight_layout()
    plt.show()
```

The distribution looks is:
![image](https://user-images.githubusercontent.com/37772157/218680350-a81425e4-7441-45c7-97e3-478fd987a045.png)

When I change the setting to `seed = 187` and $\theta =1$ I get a very similar distribution (as opposed to the paper):
![image](https://user-images.githubusercontent.com/37772157/218680824-1aec03a5-67c8-47fc-a4b1-b3a17b0da00c.png)

Where did I go wrong? I might be wrong but it looks like RBF is not applied when the data is generated and maybe that is why?

Thanks!

**Comment by strayMat** [comment]
Hi Yarin ! 

I think that the parameters of overlap 0.7 and 1 are a little too close to see something in term of overlap on these graphs. I tried with your snippet and put overlap =0.7 and overlap= 3 and had very distinct overlap populations. For information, in the main experience of the paper I varied the range of this overlap parameter from 0.1 to 3.

In the code for the figure 4, I mainly wanted to show different response surfaces functions induced by different RBF basis functions, so the overlap parameters are maybe a little too close. If you want to replicate exactly the paper figures, there is a notebook for it : https://github.com/soda-inria/caussim/blob/main/caussim/demos/caussim_examples.py

Happy to help
